src/include/miopen/tensor_reorder_util.hpp

```diff
@@ -169,26 +169,49 @@ struct GenericReorderSolution : TensorReorderAttributesBase
 inline std::unique_ptr<TensorReorderAttributesBase>
 MakeTensorReorderAttributes(const ExecutionContext& ctx_,
                             miopenDataType_t data_type_,
-                            uint32_t dim_0_,
-                            uint32_t dim_1_,
-                            uint32_t dim_2_,
-                            uint32_t dim_3_,
+                            uint64_t dim_0_,
+                            uint64_t dim_1_,
+                            uint64_t dim_2_,
+                            uint64_t dim_3_,
                             uint32_t order_0_,
                             uint32_t order_1_,
                             uint32_t order_2_,
                             uint32_t order_3_)
 {
-    std::unique_ptr<TensorReorderAttributesBase> default_ptr;
     if(!ctx_.use_hip_kernels)
     {
-        return default_ptr;
+        return nullptr;
+    }
+    ///\todo 64-bit tensor dimension
+    // Currentlly we have tensor dimension limited to 2^32 - 1
+    if((dim_0_ > UINT32_MAX) || (dim_1_ > UINT32_MAX) || (dim_2_ > UINT32_MAX) ||
+       (dim_3_ > UINT32_MAX))
+    {
+        MIOPEN_THROW("Currentlly we have tensor dimension limitation of 2^32 - 1");
```

Comment:
[Recommendation] Let's return nullptr. This is more flexible than termination, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1419#discussion_r847611475.

This can be done in later PR, of course, but https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1419#discussion_r834377391 will remain unresolved for a while in this case. Up to you.

Reply:
@aska-0096 :)

Reply:
Thanks for reminding!